python/tests/expressions/test_evaluator.py

```diff
@@ -54,110 +54,110 @@
-    assert not evaluate(Record.of(4, "a"))
+    assert evaluate(Record(a=2, b="a"))
+    assert evaluate(Record(a=3, b="a"))
+    assert not evaluate(Record(a=4, b="a"))
 
 
 def test_not_in() -> None:
     evaluate = expression_evaluator(SIMPLE_SCHEMA, NotIn("id", [1, 2, 3]), case_sensitive=True)
-    assert not evaluate(Record.of(2, "a"))
-    assert not evaluate(Record.of(3, "a"))
-    assert evaluate(Record.of(4, "a"))
+    assert not evaluate(Record(a=2, b="a"))
+    assert not evaluate(Record(a=3, b="a"))
+    assert evaluate(Record(a=4, b="a"))
 
 
 def test_is_null() -> None:
     evaluate = expression_evaluator(SIMPLE_SCHEMA, IsNull("data"), case_sensitive=True)
-    assert not evaluate(Record.of(2, "a"))
-    assert evaluate(Record.of(3, None))
+    assert not evaluate(Record(a=2, b="a"))
+    assert evaluate(Record(a=3, b=None))
 
 
 def test_not_null() -> None:
     evaluate = expression_evaluator(SIMPLE_SCHEMA, NotNull("data"), case_sensitive=True)
-    assert evaluate(Record.of(2, "a"))
-    assert not evaluate(Record.of(3, None))
+    assert evaluate(Record(a=2, b="a"))
+    assert not evaluate(Record(a=3, b=None))
 
 
 def test_is_nan() -> None:
     evaluate = expression_evaluator(FLOAT_SCHEMA, IsNaN("f"), case_sensitive=True)
-    assert not evaluate(Record.of(2, 0.0))
-    assert not evaluate(Record.of(3, float("infinity")))
-    assert evaluate(Record.of(4, float("nan")))
+    assert not evaluate(Record(a=2, b=0.0))
+    assert not evaluate(Record(a=3, b=float("infinity")))
+    assert evaluate(Record(a=4, b=float("nan")))
 
 
 def test_not_nan() -> None:
     evaluate = expression_evaluator(FLOAT_SCHEMA, NotNaN("f"), case_sensitive=True)
-    assert evaluate(Record.of(2, 0.0))
-    assert evaluate(Record.of(3, float("infinity")))
-    assert not evaluate(Record.of(4, float("nan")))
+    assert evaluate(Record(a=2, b=0.0))
+    assert evaluate(Record(a=3, b=float("infinity")))
+    assert not evaluate(Record(a=4, b=float("nan")))
 
 
 def test_not() -> None:
     evaluate = expression_evaluator(SIMPLE_SCHEMA, Not(LessThan("id", 3)), case_sensitive=True)
-    assert not evaluate(Record.of(2, "a"))
-    assert evaluate(Record.of(3, "a"))
+    assert not evaluate(Record(a=2, b="a"))
+    assert evaluate(Record(a=3, b="a"))
 
 
 def test_and() -> None:
     evaluate = expression_evaluator(SIMPLE_SCHEMA, And(LessThan("id", 3), GreaterThan("id", 1)), case_sensitive=True)
-    assert not evaluate(Record.of(1, "a"))
-    assert evaluate(Record.of(2, "a"))
-    assert not evaluate(Record.of(3, "a"))
+    assert not evaluate(Record(a=1, b="a"))
+    assert evaluate(Record(a=2, b="a"))
+    assert not evaluate(Record(a=3, b="a"))
 
 
 def test_or() -> None:
     evaluate = expression_evaluator(SIMPLE_SCHEMA, Or(LessThan("id", 2), GreaterThan("id", 2)), case_sensitive=True)
-    assert evaluate(Record.of(1, "a"))
-    assert not evaluate(Record.of(2, "a"))
-    assert evaluate(Record.of(3, "a"))
+    assert evaluate(Record(a=1, b="a"))
```

Comment:
It's a bit confusing that these names don't align with the expression that is being evaluated.

Reply:
Refactored this all of it since we now have to pass in the StructType